addons/website/static/src/js/menu/content.js

```diff
@@ -1072,8 +1072,90 @@ function _clonePage(pageId) {
     });
 }
 
+const KeyboardNavigationMixin = require('web.KeyboardNavigationMixin');
+const KeyboardNavigationWidget = Widget.extend(KeyboardNavigationMixin, {
+    /**
+     * @override
+     */
+    init() {
+        this.isFullscreen = false;
+        return KeyboardNavigationMixin.init.call(this);
+    },
+    /**
+     * @override
+     */
+    start() {
+        const ret = this._super(...arguments);
+        this.setElement($('body'));
+        return ret;
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Toggles the fullscreen mode.
+     *
+     * @private
+     * @param {Boolean} state toggle fullscreen on/off (true/false)
+     */
+    async _toggleFullscreen(state) {
+        this.isFullscreen = state;
+        const dummy = document.createElement('div');
+        const topMenu = document.querySelector('#web_editor-top-edit') || dummy;
+        const leftPanel = document.querySelector('#oe_snippets') || dummy;
+        const navbar = document.querySelector('#oe_main_menu_navbar') || dummy;
+        [topMenu, leftPanel, navbar, document.body].forEach(el => el.style.transition = 'all 0.25s');
+        if (this.isFullscreen) {
+            document.body.style.setProperty('padding', '0', 'important');
+            topMenu.style.setProperty('top', `-${window.getComputedStyle(topMenu).height}`, 'important');
+            navbar.style.setProperty('top', `-${window.getComputedStyle(navbar).height}`, 'important');
+            leftPanel.style.setProperty('left', `-${window.getComputedStyle(leftPanel).width}`, 'important');
+        } else {
+            document.body.style.removeProperty('padding');
+            topMenu.style.removeProperty('top');
+            navbar.style.removeProperty('top');
+            leftPanel.style.removeProperty('left');
+        }
```

Comment:
Maybe I am missing something but why can't this just be a class toggled on the body which will do the exact same stuff? Would be cleaner to me.

Reply:
I was thinking that it was nice to have the sizes be calculated at runtime so that it doesn't break if the size of any of those elements change, but I guess I can use translate for that. A class should work just fine and be much cleaner indeed :+1: